### PR TITLE
Update 5.4.2

### PR DESCRIPTION
**Pull request by danpetry**
# Links

[Upstream](https://git.tukaani.org/)
[Changelog](https://git.tukaani.org/?p=xz.git;a=shortlog;h=refs/tags/v5.4.2)
[Jira ticket](https://anaconda.atlassian.net/browse/PKG-1001)
[License](https://git.tukaani.org/?p=xz.git;a=blob;f=COPYING;h=e54eeacfbf3f574705c926ff403214c7973c9de0;hb=HEAD)

# Changes
- Bump version, update checksum
- Remove unused CMakeLists.txt and unused `patch` dependencies

# Further notes
- Downstream packages tested with this version on osx-arm64; no extra failing tests compared to the previous release

### CODE REVIEW

**Review by ELundby45 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   name: xz
   version: {{ version }}
 
 source:
   url: https://sourceforge.net/projects/lzmautils/files/xz-{{ version }}.tar.bz2
-  sha256: 01b71df61521d9da698ce3c33148bff06a131628ff037398c09482f3a26e5408
+  sha256: aa49909cbd9028c4666a35fa4975f9a6203ed98154fbb8223ee43ef9ceee97c3
 
 build:
   number: 1
```
Comment (ELundby45) [inline recipe/meta.yaml]:
```suggestion
  number: 0
```

Comment (ELundby45) [reply recipe/meta.yaml]: Looks good to me, minus the build number.

**Review by SC426 — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: same

Comment (danpetry) [reply recipe/meta.yaml]: Thanks for the catch.

**Pull request merged**